sensor/common/image/service_impl_test.go

```diff
@@ -0,0 +1,280 @@
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectedError:    errCentralNoReachable,
+			expectedResponse: nil,
+		},
+		"Cache miss and central is reachable": {
+			request:           createImageRequest(imageName, imageID, false),
+			notify:            common.SensorComponentEventCentralReachable,
+			expectCache:       expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:    expectRegistryHelper(suite.mockRegistryStore, 1, false),
+			expectCentralCall: expectCentralCall(suite.mockCentral, 1, createScanImageInternalResponse(imageName, imageID), nil),
+			expectedError:     nil,
+			expectedResponse:  createImageResponse(imageName, imageID),
+		},
+		"Cache miss, central is reachable and returns error": {
+			request:           createImageRequest(imageName, imageID, false),
+			notify:            common.SensorComponentEventCentralReachable,
+			expectCache:       expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:    expectRegistryHelper(suite.mockRegistryStore, 1, false),
+			expectCentralCall: expectCentralCall(suite.mockCentral, 1, nil, err),
+			expectedError:     errCentral,
+			expectedResponse:  nil,
+		},
+		"Cache miss, local scan, central is unreachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventOfflineMode,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectedError:    errCentralNoReachable,
+			expectedResponse: nil,
+		},
+		"Cache miss, local scan, central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:   expectRegistryHelper(suite.mockRegistryStore, 1, true),
+			expectLocalScan:  expectLocalScan(suite.mockLocalScan, 1, createScannedImage(imageName, imageID), nil),
+			expectedError:    nil,
+			expectedResponse: createImageResponse(imageName, imageID),
+		},
+		"Cache miss, local scan returns error, central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:   expectRegistryHelper(suite.mockRegistryStore, 1, true),
+			expectLocalScan:  expectLocalScan(suite.mockLocalScan, 1, nil, err),
+			expectedError:    errLocalScan,
+			expectedResponse: nil,
+		},
+	}
+	for testName, c := range cases {
+		suite.T().Run(testName, func(t *testing.T) {
+			suite.service.centralReady.Reset()
+			suite.service.Notify(c.notify)
+			c.expectCache.Fn()()
+			c.expectRegistry.Fn()()
+			c.expectCentralCall.Fn()()
+			c.expectLocalScan.Fn()()
+			res, err := suite.service.GetImage(ctx, c.request)
+			if c.expectedError != nil {
+				assert.EqualError(t, err, c.expectedError.Error())
```

Comment:
No need to use the static `assert` object here when using suite. Suite has one with the test pointer already embedded:
suite
```suggestion
				suite.Assert().EqualError(err, c.expectedError.Error())
```

Reply:
Done